From c170c41062518a5f5f0c4c16feda6cdc07010c75 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Wed, 11 Dec 2024 18:43:47 +0200
Subject: [PATCH] Skip Fortran tests for MSVC toolchain

Otherwise, tests fail during configuration:
  checking for Fortran 77 name-mangling scheme... ../../tests/f77demo.at:331: exit code was 1, expected 0

* tests/f77demo.at, tests/fcdemo.at: Skip tests if 'nm_interface'
is equal to "MS dumpbin".
---
diff --git a/tests/f77demo.at b/tests/f77demo.at
index 6b96920a..07a619e6 100644
--- a/tests/f77demo.at
+++ b/tests/f77demo.at
@@ -300,6 +300,9 @@ then
   AT_CHECK([echo "$LDFLAGS" | $EGREP 'fsanitize=' && (exit 77)], 1)
 fi
 
+# Skip Fortran tests for MSVC toolchain
+AT_CHECK([$LIBTOOL --config | $EGREP '^nm_interface="MS dumpbin"' && (exit 77)], 1)
+
 # Executing the static fprogram might be interactive on MSYS.
 AT_KEYWORDS([interactive])
 
@@ -326,6 +329,9 @@ then
   AT_CHECK([echo "$LDFLAGS" | $EGREP 'fsanitize=' && (exit 77)], 1)
 fi
 
+# Skip Fortran tests for MSVC toolchain
+AT_CHECK([$LIBTOOL --config | $EGREP '^nm_interface="MS dumpbin"' && (exit 77)], 1)
+
 _LT_SETUP
 
 LT_AT_CHECK_CONFIG([--disable-static],
@@ -349,6 +355,9 @@ then
   AT_CHECK([echo "$LDFLAGS" | $EGREP 'fsanitize=' && (exit 77)], 1)
 fi
 
+# Skip Fortran tests for MSVC toolchain
+AT_CHECK([$LIBTOOL --config | $EGREP '^nm_interface="MS dumpbin"' && (exit 77)], 1)
+
 _LT_SETUP
 
 LT_AT_CHECK_CONFIG([],
diff --git a/tests/fcdemo.at b/tests/fcdemo.at
index 80fe2482..9d934045 100644
--- a/tests/fcdemo.at
+++ b/tests/fcdemo.at
@@ -314,6 +314,9 @@ then
   AT_CHECK([echo "$LDFLAGS" | $EGREP 'fsanitize=' && (exit 77)], 1)
 fi
 
+# Skip Fortran tests for MSVC toolchain
+AT_CHECK([$LIBTOOL --config | $EGREP '^nm_interface="MS dumpbin"' && (exit 77)], 1)
+
 # Executing the static fprogram might be interactive on MSYS.
 AT_KEYWORDS([interactive])
 
@@ -340,6 +343,9 @@ then
   AT_CHECK([echo "$LDFLAGS" | $EGREP 'fsanitize=' && (exit 77)], 1)
 fi
 
+# Skip Fortran tests for MSVC toolchain
+AT_CHECK([$LIBTOOL --config | $EGREP '^nm_interface="MS dumpbin"' && (exit 77)], 1)
+
 _LT_SETUP
 
 LT_AT_CHECK_CONFIG([--disable-static],
@@ -363,6 +369,9 @@ then
   AT_CHECK([echo "$LDFLAGS" | $EGREP 'fsanitize=' && (exit 77)], 1)
 fi
 
+# Skip Fortran tests for MSVC toolchain
+AT_CHECK([$LIBTOOL --config | $EGREP '^nm_interface="MS dumpbin"' && (exit 77)], 1)
+
 _LT_SETUP
 
 LT_AT_CHECK_CONFIG([],
--
2.45.2

